resources/metadata/kanikoExecute.yaml

```diff
@@ -119,6 +161,10 @@ spec:
         params:
           - name: container/registryUrl
           - name: container/imageNameTag
+          - name: container/multiImageNames
+            type: "[]string"
+          - name: container/multiImageNameTags
+            type: "[]string"
```

Comment:
Shouldn't we rather introduce `container/images` or (if splitting name and tags brings a benefit) `container/imageNames` and `container/imageTags` and deprecate `container/imageNameTag`?

I don't know if there is a mechanism for such deprecations. Anyway, I don't see a reason or benefit for the `multi` prefix.

Reply:
I used `multi` as a prefix to clearly separate. We likely need the existing field and having an additional field with just an `s` at the end might easily be overlooked.

We could use `imageNames` and `imageNameTags` as well though with the slight disadvantage of overlooking the `s`
I have no strong opinion here ...

Reply:
I just don't see the `multi` as a special case (which is where our opinions might differ). 

Why would we keep the existing `container/imageNameTag` indefinitely? Every step that can *only* handle a single image needs to fail if either `container/imageNameTag` is empty or there is more than one entry in `container/images` (or however we'll end up calling the new output), right?

What about either:
```suggestion
          - name: container/imageNames
            type: "[]string"
          - name: container/imageTags
            type: "[]string"
```

or

```suggestion
          - name: container/imageNameTags
            type: "[]string"
```

or just

```suggestion
          - name: container/images # e.g. ["my-image:my-tag", "my-other-image:my-tag"]
            type: "[]string"
```

or actually, if custom types are possible or `image` is worth introducing in Piper's schema

```suggestion
          - name: container/images
            type: "[]Image"
```

where `Image` would be an object with `name`, `tag` and `reference` fields.

I'd personally go for the last if it is possible or the second last one as those keep the required information together instead of relying on `imageName[n]` and `imageTag[n]` to align well.
The second one would also fulfil that property, but would indeed only differ from the old parameter name by a single "s".
The first option would not give the automatic alignment, but the names would significantly differ from the old parameter, so if you'd go for this I also take it over the `multi` prefix.

Reply:
has been updated to `imageNames` and `imageNameTags`